src/ipc/ipc-zephyr.c

```diff
@@ -125,6 +125,12 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	/* perform command */
 	ipc_cmd(hdr);
 
+	/* are we about to enter D3 ? */
+	if (ipc->pm_prepare_D3) {
```

Comment:
@tmleman @marcinszkudlinski Any thoughts how this should be handled? This feels a bit wrong to call to platform_pm_runtime_power_off() with the native driver, but somehow we need to handle the IPC response in this case (e.g. the code in src/platform/intel/cavs/lib/power_down.S for cAVS platforms -- send IPC reply and shut down).

Reply:
I would expect here that we would have to call the Zephyr native API for shutdown() which should permit return and then allow the assembly IPC logic to send the IPC reply ? adding @nashif

Reply:
Aa, Ranjani found the logic is coming in this PR -> https://github.com/thesofproject/sof/pull/6440

Reply:
but we cannot use the same logic for TGL right @kv2019i. We need an zephyr native API for shutdown like @lgirdwood suggested. But until then my patch can be stop gap

Reply:
Ack, the right solution is of course to implement PM support for TGL as well, but in spirit of our incremental move from XTOS drivers to native drivers, having a intermediate solution here for TGL is fine.